library/core/src/iter/traits/iterator.rs

```diff
@@ -1543,11 +1543,11 @@ pub trait Iterator {
     /// collection into another. You take a collection, call [`iter`] on it,
     /// do a bunch of transformations, and then `collect()` at the end.
     ///
-    /// One of the keys to `collect()`'s power is that many things you might
-    /// not think of as 'collections' actually are. For example, a [`String`]
-    /// is a collection of [`char`]s. And a collection of
-    /// [`Result<T, E>`][`Result`] can be thought of as single
-    /// [`Result`]`<Collection<T>, E>`. See the examples below for more.
+    /// `collect()` is powerful and can also crate instances of types that
+    /// aren't typical collections. For example, a [`String`]
+    /// can be built from [`char`]s. An iterator of [`Result<T, E>`][`Result`]
```

Comment:
Using `.` here feels weird, suddenly it says

> An iterator of [`Result<T, E>`][`Result`] items can be collected into `Result<Collection<T>, E>`.

out of nowhere, it feels weird, maybe use comma or `and` here?